File: dayOne/java/replacingNonPerfectSquares/replacingNonPerfectSquares.java
public class replacingNonPerfectSquares {

    public static int[] replaceNonPerfectSquares(int[] numbers) {

        int[] replacedNonPerfectSquares = new int[numbers.length];

        for(int element = 0; element < numbers.length; element++) {

            int squareRoot = (int)Math.sqrt(numbers[element]);

            if(squareRoot * squareRoot == numbers[element]) {

                replacedNonPerfectSquares[element] = numbers[element];

            }

            else{

                replacedNonPerfectSquares[element] = -1;    
                    
            }

        }
    
        return replacedNonPerfectSquares;

    }

}

File: dayOne/python/replacing_non_perfect_squares/replacing_non_perfect_squares.py
import math

def replace_non_perfect_squares(numbers):
    replaced_non_perfect_squares = []

    for element in range(len(numbers)):
        square_root = int(math.sqrt(numbers[element]))

        if square_root * square_root == numbers[element]:
            replaced_non_perfect_squares.append(numbers[element])
        else:
            replaced_non_perfect_squares.append(-1)

    return replaced_non_perfect_squares
